### ISSUE TITLE
[tsserver] "Error processing request. watch  ENOENT" with TypeScript 1.7.3

### ISSUE THREAD

**Issue by angelozerr** [Microsoft/TypeScript#5913]
I have a problem "Error processing request. watch  ENOENT" when I send an open command. To reproduce since I have migrated to 1.7.3 (with 1.6, it worked).

Here steps to reproduce the problem:

 * create a folder `C:/sample`  which contains 2 files : 
  * `a.ts` which is empty
  *  `tsconfig.json ` which contains `{}`
 * do `cd .C:/sample`
 * do `node YOU_TYPESCRIPT_PATH/lib/tsserver.js`
 * do `{"seq":0,"type":"request","command":"open","arguments":{"file":"a.ts"}}`

After validate the open command, tsserver throws the following error:

```json
{"seq":0,"type":"response","command":"open","request_seq":0,"success":false,"mes
sage":"Error processing request. watch  ENOENT"}
```

After debugging, it seems that problem comes from https://github.com/Microsoft/TypeScript/blob/master/src/server/editorServices.ts#L1245

```javascript
project.directoryWatcher = this.host.watchDirectory(
                    ts.getDirectoryPath(configFilename),
                    path => this.directoryWatchedForSourceFilesChanged(project, path),
                    /*recursive*/ true
                );
```

The `ts.getDirectoryPath(configFilename)` return empty, and it seems that it's the problem.

Many thanks for your help.

**Comment by mhegazy** [comment]
I think all clients we had use full path for files. that is why we have not caught that. we should fix this issue, but i would recommend using the full path, this way you do not have to worry about currentDirectory.

**Comment by angelozerr** [comment]
Thanks a lot @mhegazy for your quick answer!

> I think all clients we had use full path for files

In my case I integrate tsserver inside Eclipse. Eclipse uses an instance of IProject which manages IFile instances. So the full path (filesystem path) is not interesting in this case:

 * I start tsserver inside IProject base directory.
 * I use relative path for file (relative to IProject base directory).

It's out of the topic, but it will be very cool if you could provide a recipe to implement tsClient (explain protocol with a basic sample: I open a file, I execute completion, I change the file content, etc)

**Comment by mhegazy** [comment]
i kinda tried to do that with https://github.com/Microsoft/TypeScript/blob/master/src/server/client.ts. if you have a specific topics you want to get more documentation we can start building a server doc around that.

**Comment by angelozerr** [comment]
@mhegazy I have fixed in my repository my problem https://github.com/angelozerr/typescript.java/commit/d3cd7ebbc3d0cf4b0d433a5b82d429be2666767e (but I don't know if it's the well fix)

For documentation help, I have created https://github.com/Microsoft/TypeScript/issues/5923 with my idea

**Comment by mhegazy** [comment]
I do not think this is the correct fix. the fix is to always resolve all input file path before using them.

**Comment by angelozerr** [comment]
@mhegazy I have retry with 1.7.5 and now I have the error **No Project.**

> I think all clients we had use full path for files. that is why we have not caught that. 

Ok I will try to do that.

> we should fix this issue

Or do you think when you will have time to fix it?

**Comment by mhegazy** [comment]
> Or do you think when you will have time to fix it?

not on my list at the time being :). would be happy to accept a PR for it though.

**Comment by RyanCavanaugh** [comment]
I seriously doubt this still repros in TS7 🙂
